cmd/mt-replicator/consume.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/bsm/sarama-cluster"
+	"github.com/raintank/worldping-api/pkg/log"
+	"gopkg.in/raintank/schema.v1"
+)
+
+type Consumer struct {
+	consumer *cluster.Consumer
+	Done     chan struct{}
+}
+
+func NewConsumer(brokers []string, group, topic string) (*Consumer, error) {
+	config := cluster.NewConfig()
+	// see https://github.com/raintank/metrictank/issues/236
+	config.Consumer.Offsets.Initial = sarama.OffsetNewest
+	config.ClientID = "mt-replicator"
+	config.Group.Return.Notifications = true
+	config.ChannelBufferSize = 1000
+	config.Consumer.Fetch.Min = int32(1)
+	config.Consumer.Fetch.Default = int32(32768)
```

Comment:
I don't think you need to specify the type explicitly. the compiler should be able to deduce it from the type of the attribute you're assigning to.

https://play.golang.org/p/r1VsIXSsB6